js/common.js

```diff
@@ -1374,16 +1374,53 @@ function flashIconButton(button, button_classes, icon_classes, duration) {
 }
 
 /**
- * uniqid() function, providing similar result as PHP does.
- *
- * @param {string} prefix
- * @param {boolean} random
- * @returns {string}
- *
- * @see https://stackoverflow.com/a/48593447
- */
+* uniqid() function, providing similar result as PHP does.
+*
+* @param {string} prefix
+* @param {boolean} random
+* @returns {string}
+*
+* @see https://stackoverflow.com/a/48593447
+*/
 function uniqid(prefix = "", more_entropy = false) {
    const sec = Date.now() * 1000 + Math.random() * 1000;
    const id = sec.toString(16).replace(/\./g, "").padEnd(14, "0");
    return `${prefix}${id}${more_entropy ? `.${Math.trunc(Math.random() * 100000000)}`:""}`;
 }
+
+/**
+ * 'Debounces' the click event of a button.
+ * For submit buttons, this will prevent the form from being submitted multiple times.
+ * For other buttons, this will disable them for a specified period of time based on the data-debounce-timeout attribute.
+ * If debounce-timeout is not specified, the button must be manually undebounced with the `undebounceButton` function.
+ * @param btn
+ */
+function debounceButton(btn) {
+   btn.data('original-content', btn.html());
+   btn.html(`<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>`);
+   if (btn.attr('type') === 'submit') {
+      btn.click((e) => {e.preventDefault();});
+   } else {
+      btn.prop('disabled', true);
```

Comment:
I still think that behaviour should be the same for both `submit` and `button`. Indeed, `submit` could also be used in an AJAX context (to handle submission of for using `Enter` key).
Thus, I did not tested it, but I am pretty sure that using the `Enter` key would submit the form enven if the button has been debounced.

Finally, we may have mutiple submit or regular buttons on a form (each having a different value). Maybe this function should debounce all buttons of the form, to prevent execution of another action related to the same form.

Reply:
The ability to trigger a form submission when pressing enter or ctrl-enter in form fields added by us in `common.js` just triggers a click on the `add` or `update` submit button and only if there is only one of them present. Therefore, debouncing the submit button still prevents the form submission that way.

Reply:
I'm willing to just not worry about regular buttons for this PR just to get it merged. The entire point was to prevent the confusing `The action you have requested is not allowed.` error page you get when clicking a `submit` button more than once. Having some kind of functionality to handle regular buttons was just extra.

The submit buttons cannot be disabled because that prevents even the first form submission. Regular buttons cannot just have a click handler `e => e.preventDefault()` added because that doesn't prevent anything for some reason. Being able to re-enable the submit button doesn't make sense because it should only need to be clicked once and then the page reloads. Clicking a regular button more than once makes sense because in general, they shouldn't cause a page reload. Having some kind of timeout for regular buttons would have been nice for cases where you are making AJAX calls and don't them to stack or happen so frequently.

Reply:
> The ability to trigger a form submission when pressing enter

This one is a native HTML behaviour.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event says:
> The submit event fires when the user clicks a submit button (<button> or <input type="submit">) or presses Enter while editing a field (e.g. <input type="text">) in a form. The event is not sent to the form when calling the form.submit() method directly.

But you are right, we should focus on submit buttons only for now.